docs/guides/integration-javascript.md

```diff
@@ -0,0 +1,293 @@
+
+```js
+<script defer src="{{ UID2_JS_SDK_URL }}"></script>
+ 
+<script>
+ 
+// When the UID2 SDK is executed, it will look for these callbacks and invoke them.
+window.__uid2 = window.__uid2 || {};
+window.__uid2.callbacks = window.__uid2.callbacks || [];
+window.__uid2.callbacks.push((eventType, payload) => {
+  switch (eventType) {
+    case "SdkLoaded":
+      // The SdkLoaded event occurs just once.
+      __uid2.init({});
+      break;
+ 
+    case "InitCompleted":
+      // The InitCompleted event occurs just once.
+      //
+      // If there was a valid UID2 token, it will be in payload.identity.
+      break;
+ 
+    case "IdentityUpdated":
+      // The IdentityUpdated event will happen when a UID2 token was generated or refreshed.
+      break;
+  }
+});
+ 
+</script>
+```
+
+### Using UID2 Integration Environment
+
+:::note
+By default, the SDK is configured to work with the UID2 production environment (https://prod.uidapi.com). If you want to use the UID2 integration environment instead, provide the URL for the integration environment in your call to init:
+
+```js
+__uid2.init({
+  baseUrl: "https://operator-integ.uidapi.com",
+});
+```
+
+Tokens from the UID2 integration environment are not valid for passing to the bid stream. For the integration environment, you will have different **subscription ID** and **public key** values.
+:::
+
+
+### Optional: Reduce Latency by Setting the API Base URL
+
+By default, the JS SDK makes API calls to a UID2 server in the production environment in the USA. Depending on where your users are based, you might consider choosing a server closer to your users in order to reduce latency.
+
+For example, a publisher in Singapore can set the base URL to https://sg.prod.uidapi.com. This is still the UID2 production environment, but the servers are in Singapore.
+
+For the list of possible base URLs, see [Environments](../getting-started/gs-environments.md).
+
+A publisher can also set the base URL to https://global.prod.uidapi.com. This URL will direct readers to a region geographically close to them, which is ideal if your audience is geographically distributed.
+
+To specify a different UID2 server, you can change it in the init:
+
+```js
+__uid2.init({
+  baseUrl: "https://global.prod.uidapi.com",
+});
+```
+
+## Configure the UID2 SDK For JavaScript
+
+UID2 will provide the publisher with the following values required to use the client-side token generation feature:
+
+* A subscription ID, and
+* A public key.
+
+There will be one set of these values for the publisher's testing environment(s), and a separate set for the publisher's production environment.
+
+To configure the SDK, call `setIdentityFromXXXX` with an object containing the **public key** and **subscription ID** that you received during account setup, as well as the user's hashed or unhashed [DII](../ref-info/glossary-uid.md#gl-dii) (email address or phone number).
```

Comment:
I think it would be better to list the options instead of saying `setIdentityFromXXXX`.

Reply:
Or maybe not list all four, but I don't like `setIdentityFromXXXX` much.